sympy/functions/elementary/trigonometric.py

```diff
@@ -1826,13 +1840,19 @@ def taylor_term(n, x, *previous_terms):
             return S.NegativeOne**k*euler(2*k)/factorial(2*k)*x**(2*k)
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        from sympy.functions.elementary.complexes import re
         arg = self.args[0]
         x0 = arg.subs(x, 0).cancel()
         n = (x0 + S.Pi/2)/S.Pi
         if n.is_integer:
             lt = (arg - n*S.Pi + S.Pi/2).as_leading_term(x)
             return (S.NegativeOne**n)/lt
-        return self.func(x0)
+        if x0 is S.ComplexInfinity:
+            x0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
+        if x0 in (S.Infinity, S.NegativeInfinity):
+            return AccumBounds(S.NegativeInfinity, S.Infinity)
+        return self.func(x0) if x0.is_finite else self
```

Comment:
Why is the else condition needed here (and below)?

Reply:
True I had the given a thought about this while making the latest commit here as I couldn't really find a case involving the else check but i took inspiration for ` sin, cos, tan` functions where it has been used, though I couldn't really find any test where the following is encountered

Reply:
I'll try to think of some case where that might be encountered but as of now I can't think of any !